sphinx/config.py

```diff
@@ -217,6 +217,12 @@ def convert_overrides(self, name, value):
             defvalue = self.values[name][0]
             if self.values[name][2] == Any:
                 return value
+            elif self.values[name][2] == [bool]:
+                if value in ('0', 'false', 'False'):
```

Comment:
I am a bit uneasy about this. Why not then allow `'no'` and `'off'` ? And our docs of ``-D`` would need update because it suggests  only `0` and `1` from command line (becoming when argsv is parsed `'0'` then `'1'`). Why would not `smartquotes` or `latex_use_latex_multicolumn` benefit from such extra leniency? On the other hand I have myself make the mistake to recommend using `=False` from command line... thus I can appreciate more freedom in input form but I would prefer only strict Python notation `True` and `False` in addition to `0` and `1` then. Or is there some reason you allowed `false`?

Reply:
Okay, let's start with small patterns. I'll remove `false` from the list.

>Why would not smartquotes or latex_use_latex_multicolumn benefit from such extra leniency?

This patch only modifies `latex_use_xindy`. Of course we can modify other configurations later.

Reply:
I found a way to support for config_value which having boolean as a default value, without any change. With this change,  `smartquotes` and `latex_use_latex_multicolumn` will accepts `-D` overrides automatically.